### PR TITLE
Implemented isInside() method

### PR DESCRIPTION
**Pull request by mgreminger**
I need the functionality to determine whether or not an arbitrary point is inside or outside of a solid. This pull request implements the `isInside()` method for Solid and Compound shapes by exposing `BRepClass3d_SolidClassifier` functionality for determining whether or not an arbitrary point is inside or outside of a solid or compound. Added `isInside()` method to `CQ` class as well. Added tests.

This implementation is based on `isInside()` implementation from FreeCad: https://github.com/FreeCAD/FreeCAD/blob/a56658489bc008a972aeac37cc8eeff192090ef4/src/Mod/Part/App/TopoShapePyImp.cpp#L2078-L2106

pythonocc documentation for SolidClassifier: https://cdn.rawgit.com/tpaviot/pythonocc-core/804f7f3/doc/apidoc/0.18.1/OCC.BRepClass3d.html?highlight=brepclass3d_solidclassifier#OCC.BRepClass3d.BRepClass3d_SolidClassifier

### CODE REVIEW

**Comment by mgreminger** [comment]
Thanks for the review, shouldn't be a problem for me to make the changes. I'll be able to get to it the week of the June 24th.

**Comment by adam-urbanczyk** [comment]
Thanks @mgreminger, take your time. I really appreciate your contributions to the `occ_impl` layer.

File: cadquery/occ_impl/shapes.py [hunk 1]
```diff
@@ -399,6 +400,23 @@ def centerOfMass(obj):
         else:
             raise NotImplementedError
 
+    @staticmethod
```
[the remark this answers is not in the document]
Comment (mgreminger) [reply cadquery/occ_impl/shapes.py]: Agreed, will update.

File: cadquery/cq.py [hunk 2]
```diff
@@ -2890,3 +2890,16 @@ def _repr_html_(self):
         else:
             return Compound.makeCompound(self.objects)._repr_html_()
 
+    def isInside(self, point, tolerance=1.0e-6):
```
[the remark this answers is not in the document]
Comment (mgreminger) [reply cadquery/cq.py]: Good point, not needed. Will update.

**Comment by mgreminger** [comment]
Don't merge yet. Just discovered that the current isIndside() method as it is doesn't correctly handle the case where a point is on the boundary. I'll correct and merge changes.

**Comment by mgreminger** [comment]
Should be good to merge now. I made the requested changes. In addition, it now returns true for a point on the boundary of the solid where before it returned false for a point that was within tolerance of the boundary whether or not the point was inside of the solid.

**Comment by adam-urbanczyk** [comment]
Thanks for your contribution @mgreminger! I'd like to merge it, anything against @dcowden @jmwright ?

**Comment by jmwright** [comment]
+1

**Pull request merged**